modules/core/src/lib/animation-loop-proxy.js

```diff
@@ -243,9 +247,10 @@ export default class AnimationLoopProxy {
 
   _resizeCanvasDrawingBuffer() {
     if (this.autoResizeDrawingBuffer) {
-      const devicePixelRatio = this.useDevicePixels ? window.devicePixelRatio || 1 : 1;
-      const width = this.canvas.clientWidth * devicePixelRatio;
-      const height = this.canvas.clientHeight * devicePixelRatio;
+      // TODO: get access to gl context and use 'mapToDevicePosition'
+      const devicePixelRatio = getDevicePixelRatio(this.useDevicePixels, window.devicePixelRatio);
```

Comment:
What do we gain by passing in the value of the global `window.devicePixelRatio` in every invocation to `getDevicePixelRatio`, as opposed to referencing it inside `getDevicePixelRatio`?

Reply:
Intention was to de-couple usages `window` inside this utility method, so this can be used in non window (node) environments, also we already have reduced accessing `window.devicePixelRatio` to very few locations in the code base.

Reply:
> Intention was to de-couple usages window inside this utility method, so this can be used in non window (node) environments, 

In that case, it would still seem helpful to move the `window` reference inside `getDevicePixelRatio`, so that we only need to check for `typeof window === 'undefined'` in one place...

Reply:
Done.